src/main/java/com/rtb/projectmanagementtool/auth/AuthOps.java

```diff
@@ -61,22 +67,29 @@ public void loginUser(HttpServletRequest request, HttpServletResponse response)
     }
   }
 
+  // returns the userID long (or -1) associated with the user logged in
   public long whichUserLoggedIn(HttpServletRequest request, HttpServletResponse response) {
-    Cookie currCookie = new Cookie(COOKIENAME, NO_LOGGED_IN_USER);
-
-    // get all cookies from request
-    Cookie[] cookies = request.getCookies();
-    if (cookies != null) {
-      for (Cookie cookie : cookies) {
-        if (cookie.getName().equals("sessionUserID")) {
-          // if we find cookie w/ name, set currCookie equal
-          currCookie.setValue(cookie.getValue());
-        }
-      }
-    }
+    currCookie = generateCurrCookie(request, response);
 
     // get cookie value for user
     String currUserIDString = currCookie.getValue();
     return Long.parseLong(currUserIDString);
   }
+
+  // logs out user by setting cookie value to -1
+  public void logoutUser(HttpServletRequest request, HttpServletResponse response) {
+    currCookie = generateCurrCookie(request, response);
```

Comment:
We don't need to check for the current cookie in the request for logout.  We can just blindly always set to NO_LOGGED_IN_USER on the response.  This would be like if  you click Logout while you're logged in, vs. if you click Logout while you're not currently logged in, they would (should) probably behave the same way.